scripts/jenkins/cloud/ansible/roles/list_packages/tasks/diff-packages.yml

```diff
@@ -0,0 +1,54 @@
+#
+# (c) Copyright 2019 SUSE LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+#
+---
+
+- name: Diff list of installed packages in plaintext
+  shell: |
+    diff -U 0 /tmp/rpms/rpms_{{ item }}_{{ state1 }}.txt \
+      /tmp/rpms/rpms_{{ item }}_{{ state2 }}.txt \
+      > /tmp/rpms/diffs/rpms_{{ item }}.diff
```

Comment:
instead of shelling you can cleverly use the copy module with `check_mode: yes` so it actually doesn't copy anything and the `diff: yes` to get the changes in the diff, output is a bit crappy thougth, not sure if it would work properly for you:

```

    changed: true
    diff:
    - after: |-
        dfnusdf
        sdf
        sdf
        sd
        f
      after_header: /home/itxaka/test.txt
      before: |-
        asfdasd
      before_header: test1.txt
    failed: false
```

Reply:
Thx this is nice. I used it to print diff on the console but it didn t work well when I wanted to print that into file(artifact).